runtime/src/bank/partitioned_epoch_rewards/calculation.rs

```diff
@@ -298,28 +298,14 @@ impl Bank {
             cached_vote_accounts,
         } = reward_calculate_params;
 
-        let solana_vote_program: Pubkey = solana_vote_program::id();
-
-        let get_vote_account = |vote_pubkey: &Pubkey| -> Option<VoteAccount> {
-            if let Some(vote_account) = cached_vote_accounts.get(vote_pubkey) {
-                return Some(vote_account.clone());
-            }
-            // If accounts-db contains a valid vote account, then it should
-            // already have been cached in cached_vote_accounts; so the code
-            // below is only for sanity checking, and can be removed once
-            // the cache is deemed to be reliable.
-            metrics.vote_accounts_cache_miss_count.fetch_add(1, Relaxed);
-            let account = self.get_account_with_fixed_root(vote_pubkey)?;
-            VoteAccount::try_from(account).ok()
-        };
-
         let new_warmup_cooldown_rate_epoch = self.new_warmup_cooldown_rate_epoch();
         let estimated_num_vote_accounts = cached_vote_accounts.len();
         let vote_account_rewards: VoteRewards = DashMap::with_capacity_and_hasher(
             estimated_num_vote_accounts,
             AHashRandomState::default(),
         );
         let total_stake_rewards = AtomicU64::default();
+        const ASSERT_STAKE_CACHE: bool = false;
```

Comment:
@jeffwashington 
I added a const flag which we can toggle to assert that we will never find an vote account's that in accounts-db but not in stake cache. Is this similar to what you had in your mind?

Reply:
i'm also fine if you cut this out before merging.